### PR TITLE
refactor(scheduling): use react hooks and react query, phase out redux

### LINKED ISSUE #2338: Refactor scheduling module to use react query
## 🚀 Feature Proposal
Refactor the scheduling module to to use react query instead of redux

Need 5 hooks
- useAppointment
- useAppointments
- useScheduleAppointment
- useUpdateAppointment
- useDeleteAppointment

Refactor the code to use these new hooks

## Motivation
Remove redux

## Technical Notes
- use incidents module for inspiration
- remove redux after completion

### PR DESCRIPTION
**Pull request by WinstonPoh**
Fixes #2338 

**Changes proposed in this pull request:**

- refactored scheduling/appointments to use react hooks instead of redux
- added tests
- removed appointment(s)-slice and their tests

### CODE REVIEW

**Comment by tehKapa** [comment]
@WinstonPoh there are some build errors

```
Failed to compile.
14

15
./src/scheduling/appointments/ViewAppointments.tsx
16
  Line 72:6:  React Hook useEffect has a missing dependency: 'appointments'. Either include it or remove the dependency array  react-hooks/exhaustive-deps
17

18
./src/scheduling/appointments/new/NewAppointment.tsx
19
  Line 72:6:  React Hook useEffect has missing dependencies: 'history', 'isErrorNewAppointment', 'newAppointment', 'newAppointmentMutate', and 't'. Either include them or remove the dependency array  react-hooks/exhaustive-deps
20

21
./src/scheduling/appointments/view/ViewAppointment.tsx
22
  Line 86:6:  React Hook useCallback has missing dependencies: 'history' and 't'. Either include them or remove the dependency array  react-hooks/exhaustive-deps
23
  Line 94:6:  React Hook useEffect has a missing dependency: 'setButtonToolBar'. Either include it or remove the dependency array     react-hooks/exhaustive-deps
```

**Comment by WinstonPoh** [comment]
@tehKapa thanks for the heads up, looking into it

**Review by tehKapa — APPROVED** [review]

File: src/__tests__/scheduling/appointments/edit/EditAppointment.test.tsx [hunk 1]
```diff
@@ -55,21 +53,23 @@ describe('Edit Appointment', () => {
   let history: any
   let store: MockStore
 
-  const setup = async () => {
+  const setup = async (mockAppointment: Appointment, mockPatient: Patient) => {
+    jest.resetAllMocks()
     jest.spyOn(titleUtil, 'useUpdateTitle').mockImplementation(() => jest.fn())
+    // jest.spyOn(titleUtil, 'useUpdateAppointment').mockImplementation(() => jest.fn())
```
Comment (blestab) [inline src/__tests__/scheduling/appointments/edit/EditAppointment.test.tsx:59]: Let's remove commended out code

File: src/__tests__/scheduling/appointments/edit/EditAppointment.test.tsx [hunk 2]
```diff
@@ -94,32 +94,28 @@ describe('Edit Appointment', () => {
   })
 
   it('should render an edit appointment form', async () => {
-    const { wrapper } = await setup()
+    const { wrapper } = await setup(expectedAppointment, expectedPatient)
 
-    expect(wrapper.find(AppointmentDetailForm)).toHaveLength(1)
+    expect(wrapper.find(Spinner)).toHaveLength(1)
```
Comment (blestab) [inline src/__tests__/scheduling/appointments/edit/EditAppointment.test.tsx:99]: I am not sure i understand how finding the Spinner confirms the presence of the AppointmentDetailForm

File: src/__tests__/scheduling/appointments/view/ViewAppointment.test.tsx [hunk 3]
```diff
@@ -161,115 +156,99 @@ describe('View Appointment', () => {
...
-    })
-
-    it('should pop up the modal when on delete appointment click', async () => {
-      const { wrapper } = await setup('completed', [
-        Permissions.ReadAppointments,
-        Permissions.DeleteAppointment,
-      ])
+  it('should render a delete appointment button in the button toolbar', async () => {
+    await setup('completed', [Permissions.ReadAppointments, Permissions.DeleteAppointment])
+    // expect(setButtonToolBarSpy).toHaveBeenCalledTimes(1)
```
Comment (blestab) [inline src/__tests__/scheduling/appointments/view/ViewAppointment.test.tsx:161]: Let's remove unused/commented out code

File: src/__tests__/scheduling/appointments/view/ViewAppointment.test.tsx [hunk 4]
```diff
@@ -161,115 +156,99 @@ describe('View Appointment', () => {
...
+      Permissions.ReadAppointments,
+      Permissions.DeleteAppointment,
+    ])
 
-      act(() => {
-        const { onClick } = (actualButtons[0] as any).props
-        onClick({ preventDefault: jest.fn() })
-      })
-      wrapper.update()
+    // expect(setButtonToolBarSpy).toHaveBeenCalledTimes(1)
```
Comment (blestab) [inline src/__tests__/scheduling/appointments/view/ViewAppointment.test.tsx:174]: Let's remove unused/commented out code

File: src/__tests__/scheduling/appointments/view/ViewAppointment.test.tsx [hunk 5]
```diff
@@ -161,115 +156,99 @@ describe('View Appointment', () => {
...
```
Comment (blestab) [inline src/__tests__/scheduling/appointments/view/ViewAppointment.test.tsx:193]: Let's remove commented out code

**Review by blestab — CHANGES_REQUESTED** [review]

File: src/__tests__/scheduling/appointments/edit/EditAppointment.test.tsx (hunk 1 above)
Comment (WinstonPoh) [reply src/__tests__/scheduling/appointments/edit/EditAppointment.test.tsx:59]: 👍🏼

File: src/__tests__/scheduling/appointments/edit/EditAppointment.test.tsx (hunk 2 above)
Comment (WinstonPoh) [reply src/__tests__/scheduling/appointments/edit/EditAppointment.test.tsx:99]: Whoops yeah, it really doesn't.

**Review by blestab — APPROVED** [review]

**Comment by WinstonPoh** [comment]
Hi @tehKapa did you leave some review comments that I missed? If you did I could not find them. Is there anything more to be done on my end for now?

**Review by tehKapa — APPROVED** [review]

**Pull request merged**